python/pyarrow/tests/test_python.py

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Small nit, but maybe call this `test_cpp.py` or `test_pyarrow_cpp.py` instead? ("python" is a bit generic)

Reply:
+1, or also `test_cpp_internals.py`

Reply:
Yes, that sounds good

Reply:
OK, will do 👍